ui/actions/nuxeo-share-button.js

```diff
@@ -136,28 +148,39 @@ import './nuxeo-action-button-styles.js';
 
     _toggleDialog() {
       this.$.dialog.toggle();
-      this._selectPermalink();
     }
 
     _buildPermalink(document) {
       return document ? `${window.location.origin + window.location.pathname}#!/doc/${document.uid}` : '';
     }
 
-    _copyPermalink() {
-      this._selectPermalink();
+    _copyLink(e) {
+      const button = e.currentTarget;
+      const link = button.previousElementSibling;
+
+      this._selectLink(link);
       if (!window.document.execCommand('copy')) {
         return;
       }
 
-      const link = this.$.copyLink;
-      this._debouncer = Debouncer.debounce(this._debouncer, timeOut.after(3000), () => {
-        link.innerText = this.i18n('shareButton.operation.copy');
+      button._debouncer = Debouncer.debounce(button._debouncer, timeOut.after(2000), () => {
```

Comment:
Do we need to change the timeout value here?

Reply:
It was suggested by Carolina to reduce the time because three seconds seems a lot.